### ISSUE TITLE
No change made in Route 53

### ISSUE THREAD

**Issue by mostlyAtNight** [kubernetes/kops#5037]
I have been scratching my head for hours yesterday trying to get this to work:

TL;DR: I set up the plugin according to the guide inc. permissions.  When I create my service/deployment, no Route53 records are created and there's no error message.  

Please see details below.

Thanks for submitting an issue! Please fill in as much of the template below as
you can.

------------- BUG REPORT TEMPLATE --------------------

1. What `kops` version are you running? The command `kops version`, will display
 this information.

Version 1.8.1 (git-94ef202)

2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.

Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-03-26T16:55:54Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.7", GitCommit:"b30876a5539f09684ff9fde266fda10b37738c9c", GitTreeState:"clean", BuildDate:"2018-01-16T21:52:38Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}

3. What cloud provider are you using?

AWS

4. What commands did you run?  What is the simplest way to reproduce this issue?

1. Set up kops as described [here](https://github.com/kubernetes/kops/blob/master/docs/aws.md) (gossip DNS)
2. Install Route 53 plugin as described [here](https://github.com/kubernetes/kops/tree/master/addons/route53-mapper)
3. Added the following permissions to my kops config:

`spec:
  additionalPolicies:
    master: |
      [
        {
          "Effect": "Allow",
          "Action": ["route53:ListHostedZonesByName"],
          "Resource": ["*"]
        },
        {
          "Effect": "Allow",
          "Action": ["elasticloadbalancing:DescribeLoadBalancers"],
          "Resource": ["*"]
        },
        {
          "Effect": "Allow",
          "Action": ["route53:ChangeResourceRecordSets"],
          "Resource": ["*"]
        }
      ]
    node: |
      [
        {
          "Effect": "Allow",
          "Action": ["route53:ListHostedZonesByName"],
          "Resource": ["*"]
        },
        {
          "Effect": "Allow",
          "Action": ["elasticloadbalancing:DescribeLoadBalancers"],
          "Resource": ["*"]
        },
        {
          "Effect": "Allow",
          "Action": ["route53:ChangeResourceRecordSets"],
          "Resource": ["*"]
        }
      ]
  api:
`
4. Ran kops update cluster kops-sandpit.k8s.local --yes --state s3://kops-sandpit-lutraconsulting-co-uk-state-store
5. Ran kops rolling-update cluster --state --state s3://kops-sandpit-lutraconsulting-co-uk-state-store
6. Ran kubectl create -f kube.kops.yaml

Content of the yaml file:

`apiVersion: v1
kind: Service
metadata:
  name: mappin-web
  annotations:
    #
    # LOAD BALANCERS AND CERTIFICATES
    # The certificate to use (make this in AWS certificate management)
    service.beta.kubernetes.io/aws-load-balancer-ssl-cert: arn:aws:acm:eu-central-1:433835555346:certificate/13627a4c-5055-4346-b3f7-237528be1046
    # the ELB speaks http to the pods
    service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http
    # Only speak SSL to connections on port 443, anything else would get forwarded (e.g. 80)
    service.beta.kubernetes.io/aws-load-balancer-ssl-ports: "443"
    # https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/elb-security-policy-table.html
    service.beta.kubernetes.io/aws-load-balancer-ssl-negotiation-policy: "ELBSecurityPolicy-2016-08"
    #
    # DNS
    domainName: "mappin.lutraconsulting.co.uk"
  labels:
    app:  mappin
    role: web
    tier: frontend
    # Required by Route53 Mapping Service to handle DNS automagically
    # https://github.com/kubernetes/kops/tree/master/addons/route53-mapper
    dns: route53
spec:
  type: LoadBalancer
  ports:
  - port: 443
    targetPort: 80
    protocol: TCP
    name: https
  selector:
    app: mappin
    role: web
    tier: frontend
---
apiVersion: apps/v1beta2 #  for k8s versions before 1.9.0 use apps/v1beta2  and before 1.8.0 use extensions/v1beta1
kind: Deployment
metadata:
  name: mappin-web
spec:
  selector:
    matchLabels:
      app:  mappin
      role: web
      tier: frontend
  replicas: 1
  template:
    metadata:
      labels:
        app: mappin
        role: web
        tier: frontend
    spec:
      containers:
      - name: web
        image: 433835555346.dkr.ecr.eu-central-1.amazonaws.com/mappin_web:latest
        # imagePullPolicy: Never # local, do not download from hub
        resources:
          requests:
            cpu: 0.1
            memory: 100Mi
        ports:
        - containerPort: 80
`

(output was:)
  service "mappin-web" created
  deployment.apps "mappin-web" created

5. What happened after the commands executed?

output was:
  service "mappin-web" created
  deployment.apps "mappin-web" created

No new entries were created in Route 53

6. What did you expect to happen?

A new entry called mappin.lutraconsulting.co.uk to be created in Route 53

7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -oyaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.

apiVersion: kops/v1alpha2
kind: Cluster
metadata:
  creationTimestamp: 2018-04-06T03:57:37Z
  name: kops-sandpit.k8s.local
spec:
  additionalPolicies:
    master: |
      [
        {
          "Effect": "Allow",
          "Action": ["route53:ListHostedZonesByName"],
          "Resource": ["*"]
        },
        {
          "Effect": "Allow",
          "Action": ["elasticloadbalancing:DescribeLoadBalancers"],
          "Resource": ["*"]
        },
        {
          "Effect": "Allow",
          "Action": ["route53:ChangeResourceRecordSets"],
          "Resource": ["*"]
        }
      ]
    node: |
      [
        {
          "Effect": "Allow",
          "Action": ["route53:ListHostedZonesByName"],
          "Resource": ["*"]
        },
        {
          "Effect": "Allow",
          "Action": ["elasticloadbalancing:DescribeLoadBalancers"],
          "Resource": ["*"]
        },
        {
          "Effect": "Allow",
          "Action": ["route53:ChangeResourceRecordSets"],
          "Resource": ["*"]
        }
      ]
  api:
    loadBalancer:
      type: Public
  authorization:
    alwaysAllow: {}
  channel: stable
  cloudProvider: aws
  configBase: s3://kops-sandpit-lutraconsulting-co-uk-state-store/kops-sandpit.k8s.local
  etcdClusters:
  - etcdMembers:
    - instanceGroup: master-eu-central-1a
      name: a
    name: main
  - etcdMembers:
    - instanceGroup: master-eu-central-1a
      name: a
    name: events
  iam:
    allowContainerRegistry: true
    legacy: false
  kubernetesApiAccess:
  - 0.0.0.0/0
  kubernetesVersion: 1.8.7
  masterInternalName: api.internal.kops-sandpit.k8s.local
  masterPublicName: api.kops-sandpit.k8s.local
  networkCIDR: 172.20.0.0/16
  networking:
    kubenet: {}
  nonMasqueradeCIDR: 100.64.0.0/10
  sshAccess:
  - 0.0.0.0/0
  subnets:
  - cidr: 172.20.32.0/19
    name: eu-central-1a
    type: Public
    zone: eu-central-1a
  topology:
    dns:
      type: Public
    masters: public
    nodes: public

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-04-06T03:57:38Z
  labels:
    kops.k8s.io/cluster: kops-sandpit.k8s.local
  name: master-eu-central-1a
spec:
  image: kope.io/k8s-1.8-debian-jessie-amd64-hvm-ebs-2018-02-08
  machineType: m3.medium
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-eu-central-1a
  role: Master
  subnets:
  - eu-central-1a

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-04-06T03:57:38Z
  labels:
    kops.k8s.io/cluster: kops-sandpit.k8s.local
  name: nodes
spec:
  image: kope.io/k8s-1.8-debian-jessie-amd64-hvm-ebs-2018-02-08
  machineType: t2.medium
  maxSize: 2
  minSize: 2
  nodeLabels:
    kops.k8s.io/instancegroup: nodes
  role: Node
  subnets:
  - eu-central-1a

8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.

[err.log](https://github.com/kubernetes/kops/files/1926205/err.log)

9. Anything else do we need to know?

Don't think so.

------------- FEATURE REQUEST TEMPLATE --------------------

1. Describe IN DETAIL the feature/behavior/change you would like to see.

2. Feel free to provide a design supporting your feature request.

**Comment by huang-jy** [comment]
Basic troubleshooting checks.

1. Do you run with more than one aws profile?
2. Are the AWS_ACCESS_KEY and AWS_SECRET_ACCESS_KEY variables populated?
3. Are you using the correct profile
4. Can the profile you're using actually view or see the Route 53 zone?
5. (I've done this) -- have you forgotten to change your profile and actually created the entries in the wrong account? :man_facepalming:
